bigbang/archive.py

```diff
@@ -248,6 +250,34 @@ def compute_activity(self, clean=True):
 
         return activity
 
+    def get_froms(self):
+        """
+        Returns a dataframe with a row for every message of the archive, containing
+        column entries for:
+
+         - The From field of the email
+         - The email address extracted from the From field
+         - The domain of the From field
+
+        This dataframe is computed the first time this method is called and then cached.
+
+        Returns
+        ----------
+
+        froms: pandas.DataFrame
+        """
+        if self.froms is not None:
+            return self.froms
+        else:
+            emails = self.data["From"].apply(domain.extract_email)
```

Comment:
In this instance, to get emails and domains, one has to loop twice over all DataFrame rows. Could we speed this up by using only one loop?